### DISCUSSION TITLE
Utility of PCSC mode

### DISCUSSION THREAD

**Discussion by mhalano** [Yubico/libfido2#626] [Q&A] [answered]
I saw in the latest release of libfido2 it has a flag to enable a experimental PCSC support. My question is what exactly is this feature?

**Comment by martelletto** [comment, marked as answer]
libfido2's PCSC backend allows the library to communicate with U2F/FIDO2 devices that are exposed by the system's PCSC interface. These include NFC devices on Linux that are not supported by the kernel's NFC driver (see https://ccid.apdu.fr/ccid/supported.html), and NFC devices in general on macOS and Windows (modulo Microsoft's webauthn.dll, which we also support, but handles PCSC internally).

**Reply by AdityaMitra5102** [reply]
I am trying to perform webauthn from a linux device (Raspberry Pi) over NFC but my browsers (Chromium and Firefox) are expecting me to insert the Security Key by USB and not reading when I tap it on the NFC Reader (Omnikey 5022). What could be the solution?

**Comment by justin0mcateer** [comment]
I desperately need to do the same thing, but
I don't think it is possible on Chrome currently.

I am planning to test the same scenario on Windows today.


________________________________

**Reply by LDVG** [reply]
Note that Chrome and Firefox use their own implementation for communicating with authenticators, libfido2 is not involved. As far as I know, NFC is currently unsupported on Linux in those browsers.

CC: @AdityaMitra5102

**Reply by AdityaMitra5102** [reply]
Okay. Just askin, is there any known browser that supports NFC on Linux?

**Reply by LDVG** [reply]
Not that I'm aware of.

**Reply by GeorgFleig** [reply]
FYI: There is a workaround available that emulates a USB device and forwards the FIDO2 communication via PCSC to the YubiKey. With this I got FIDO2 working on Linux with a NFC reader using Firefox and Chrome.

These are two implementations of the workaround that I am aware of:
- https://github.com/BryanJacobs/fido2-hid-bridge
- https://github.com/StarGate01/CTAP-bridge (may require custom kernel or additional modules, depending on your distro)

**Comment by justin0mcateer** [comment]
FWIW, it works without issue using Chrome on Windows.


________________________________

**Reply by AdityaMitra5102** [reply]
Yes. I personally use Edge on Windows and it works perfectly. It uses the internal webauthn handler of Windows which is available at C:/Windows/System32/webauthn.dll

**Comment by justin0mcateer** [comment]
Thanks for that info!

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
